kaggle_environments/core.py

```diff
@@ -643,21 +648,25 @@ def act(none_action=None):
         return structify({"act": act})
 
     def __get_shared_state(self, position):
-        if position == 0:
-            return self.state[0]
-        state = copy.deepcopy(self.state[position])
-
         # Note: state and schema are required to be in sync (apart from shared ones).
         def update_props(shared_state, state, schema_props):
             for k, prop in schema_props.items():
-                if get(prop, bool, path=["shared"], fallback=False):
+                # Hidden fields are tracked in the episode replay but are not provided to the agent at runtime
+                if get(prop, bool, path=["hidden"], fallback=False):
+                    if k in state:
```

Comment:
Yes it's allowed, for shared hidden properties (like thresholds), `k` will only appear on the first agent's observation since all agents share that property.